src/store/types.ts

```diff
@@ -1,11 +1,14 @@
 import { IADSApiBootstrapData } from '@api';
+import { Theme } from '../types';
 
 export interface IAppState {
   user: IADSApiBootstrapData;
+  theme: Theme;
 }
 
 export enum AppEvent {
   SET_USER = 'SET_USER',
+  SET_THEME = 'SET_THEME',
 }
 
-export type Action = { type: AppEvent.SET_USER; payload: IAppState['user'] };
+export type Action = { type: AppEvent; payload: IAppState['user'] | IAppState['theme'] };
```

Comment:
Lets switch to using discriminated unions here, it'll type better (https://basarat.gitbook.io/typescript/type-system/discriminated-unions#discriminated-union)


```suggestion
export type Action = 
| { type: AppEvent.SET_USER; payload: IAppState['user'] }
| { type: AppEvent.SET_THEME; payload: IAppState['theme'] }
```

Reply:
Once you make this change you should be able to go back to the areas you've used them and remove any casting (i.e. `as Theme`)